build/DeployTasks/DeployVsixToMarketplaceTask.cs

```diff
@@ -0,0 +1,57 @@
+﻿
+using System.Net.Http.Headers;
+using System.Text;
+
+namespace BuildScripts;
+
+[TaskName("DeployVsixToMarketplaceTask")]
+[IsDependentOn(typeof(DownloadArtifactsTask))]
+public sealed class DeployVsixToMarketplaceTask : FrostingTask<BuildContext>
+{
+    public override bool ShouldRun(BuildContext context)
+    {
+        if (context.BuildSystem().IsRunningOnGitHubActions)
+        {
+            var workflow = context.BuildSystem().GitHubActions.Environment.Workflow;
+            if (workflow.RefType == GitHubActionsRefType.Tag &&
+                !string.IsNullOrWhiteSpace(context.EnvironmentVariable("MARKETPLACE_PAT")))
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public override void Run(BuildContext context)
```

Comment:
You should be able to make this an `async void Run()`

Reply:
I wasn't sure how much I could do of that. I didn't want to break anything. Done.